runtime/src/snapshot_bank_utils.rs

```diff
@@ -579,8 +579,13 @@ fn verify_slot_deltas(
     slot_deltas: &[BankSlotDelta],
     bank: &Bank,
 ) -> std::result::Result<(), VerifySlotDeltasError> {
-    let info = verify_slot_deltas_structural(slot_deltas, bank.slot())?;
-    verify_slot_deltas_with_history(&info.slots, &bank.get_slot_history(), bank.slot())
+    let slot = bank.slot();
+    let info = verify_slot_deltas_structural(slot_deltas, slot)?;
+    let slot_history = &bank.get_slot_history();
+    // ensure the slot history is valid (as much as possible), since we're
+    // using it to verify the slot deltas
+    verify_slot_history(slot_history, slot)?;
```

Comment:
nit: I would have put this inside `fn verify_slot_deltas_with_history` so that we have a stronger guarantee that slot history is verified before verifying slot deltas (ie some new callsite doesn't forget to verify slot history)